### PR TITLE
Rename utilities like self-center to align-self-center

### PR DESCRIPTION
**Pull request by adamwathan**
Since there is also a `justify-self` property, keeping the existing `self-center` style class names means there is no good name for `justify-self` utilities if we introduce them later. This PR changes those class names to be slightly more verbose so that there is no ambiguity.

| Old | New |
| --- | --- |
| `self-auto` | `align-self-auto` |
| `self-start` | `align-self-start` |
| `self-end` | `align-self-end` |
| `self-center` | `align-self-center` |
| `self-stretch` | `align-self-stretch` |

### CODE REVIEW

**Comment by benface** [comment]
@adamwathan There is also a `justify-items` property, does that mean `items-start` should become `align-items-start`, etc.?

**Comment by adamwathan** [comment]
Ugh that's grim I really don't want to change those. Hmph, now not sure what to do.

**Comment by adamwathan** [comment]
Going to just change nothing here, and instead if we add `justify-self` and `justify-items` utilities we can make those names more verbose to disambiguate without an annoying BC break here.

| CSS Property | Class |
| --- | --- |
| `justify-content` | `justify-*` |
| `align-items` | `items-*` |
| `align-content` | `content-*` |
| `align-self` | `self-*` |
| `justify-items` | `justify-items-*` |
| `justify-self` | `justify-self-*` |

Basically the Flexbox properties get preferential treatment and get the shorter names, and the CSS Grid-only properties are longer to disambiguate themselves.

**Comment by hacknug** [comment]
> Hmph, now not sure what to do.

Here we go again 😂

Here's my two (hundred) cents:

- You can `align`, `justify`, and `place` things
- Things can refer to either `content`, `items`, or `self`
- Values for the alignment vary between properties

Imho this looks pretty similar to the [`margin`](https://tailwindcss.com/docs/spacing), [`padding`](https://tailwindcss.com/docs/spacing), and [`borderWidth`](https://tailwindcss.com/docs/border-width) plugins where we have a `base`, a `side`, and a `modifier`. Instead of `side`, what we're dealing with here is one of those _things_ I mentioned earlier.

If we want to support CSS Grid at some point, we're gonna have to deal with this somehow. Best options I see force us to break more things in v1:
- Verbose à la `borderWidth`: `.align-content-last-baseline`
- Shorter à la `padding`: `.ac-last-baseline`

I'd go with the shorter option since we're all already `.flex.justify-center.items-center` all the time. In this case we'd end up with `.flex.jc-center.ai-center`. Looks pretty weird but I feel it's going to be a better way to handle it than having those super long class names down the road (wasn't this the reason why `margin` and `padding` got the special treatment?).

Also, if we're ever going to support logical properties (which we should at some point to better support non-rtl-ttb languages :pray:) the shorter convention would have the potential for messing things up. `.pi-{{ whatever }}` could stand for wither `padding-inline` or `place-items`. Not a big deal since probably not many people will name their paddings `center`, `start`, or similar (I guess?).

Of course doing this could also mean changing the whole `verticalAlignment` plugin classes just because of `baseline`, so… 🤷‍♂️

**Comment by adamwathan** [comment]
Yeah good thoughts, the short version would be nice but I'm really just hesitant to make big breaking changes if we can avoid it. Maybe for the next CSS framework, haha...

**Comment by benface** [comment]
For what it's worth, I tend to agree with @hacknug on this. I feel like Grid will probably be just as commonly used as Flexbox in the future, and according to the Box Alignment spec, `justify-items` can also be used in block-level layouts and on absolutely-positioned elements. I totally understand not wanting to change this so close to the v1 release as it's a breaking change, but I think it will be inevitable at some point... maybe for Tailwind v2?

**Comment by adamwathan** [comment]
Yeah maybe for v2 if everyone agrees it's a real pain point. Personally I'm not convinced many people will be using `justify-items` or `justify-self` even with CSS Grid — they just seem like "really useful in the 0.001% of times you need them but usually you don't need them" sorts of features.

**Comment by PutziSan** [comment]
I use `display: grid` and the different `justify-*`/`align-*` CSS properties quite often and have therefore adapted the config for me so that the names are longer but more accurate. Since I use auto-complete of my IDE anyway, the long names are not so important to me.
If it's interesting for someone else, I've added the config as a [public gist](https://gist.github.com/PutziSan/0c9d0b94d3adb16de188cdfc0b7ef110)
